File: php/beer-song/beer-song.php
<?php
// Bof …
class BeerSong{
    function verse(int $i){
        $verse = [
            ucfirst("{$this->bottles($i)} on the wall, {$this->bottles($i)}."),
            ucfirst("{$this->take($i)}, {$this->bottles($i-1)} on the wall."),
        ];
        return implode("\n",$verse) . (($i > 0) ? "\n" : "");
    }

    function bottles(int $i){
        $i = $i >= 0 ? $i : 99;
        return (
            ($i > 0) ? $i : "no more"
        ) . " bottle" . (
            ($i > 1 || $i == 0) ? "s" : ""
        ) . " of beer";
    }

    function take(int $i){
        if($i > 0){
            return "take " . ( ($i > 1) ? "one" : "it" ) . " down and pass it around";
        }else{
            return "go to the store and buy some more";
        }
    }

    function verses(int $s, int $e){
        $verses = [];
        foreach(range($s,$e) as $i){
            $verses[] = $this->verse($i);
        }
        return implode("\n", $verses);
    }

    function lyrics(){
        return $this->verses(99,0);
    }
}

if(!debug_backtrace()){
    $s = new BeerSong();
    print($s->verses(99,98));
}

File: python/beer-song/beer_song.py
def recite(start, take=1):
    verses = []
    for i in range(take):
        k = start - i
        v = verse(k)
        verses += v + [""]
    return verses[0:-1]


def verse(i):
    verse = [
        "{b} on the wall, {b}.".format(b=bottles(i)).capitalize(),
        "{t}, {b} on the wall.".format(t=take(i), b=bottles(i-1)).capitalize(),
    ]
    return verse


def bottles(i):
    i = i if i >= 0 else 99
    return "{q} bottle{s} of beer".format(
        q=i or "no more",
        s="s" if (i > 1) or (i == 0) else ""
    )


def take(i):
    if i > 0:
        return "take {q} down and pass it around".format(
            q="one" if i > 1 else "it"
        )
    else:
        return "go to the store and buy some more"
